charmcraft/application/commands/lifecycle.py

```diff
@@ -173,13 +173,46 @@ def run_managed(self, parsed_args: argparse.Namespace) -> bool:
         # notably here, the bundle plugin doesn't work on Windows.
         if sys.platform == "linux" and charmcraft_yaml and charmcraft_yaml.get("type") == "bundle":
             return False
+
         return super().run_managed(parsed_args)
 
+    def _update_charm_libs(self) -> None:
+        """Update charm libs attached to the project."""
+        craft_cli.emit.progress("Checking that charmlibs match 'charmcraft.yaml' values")
+        project = cast(models.CharmcraftProject, self._services.project)
+        libs_svc = cast(services.CharmLibsService, self._services.charm_libs)
+        installable_libs: list[models.CharmLib] = []
+        for lib in project.charm_libs:
+            charm_name, _, lib_name = lib.lib.partition(".")
+            if not libs_svc.get_lib_version(charm_name=charm_name, lib_name=lib_name):
+                installable_libs.append(lib)
+        if installable_libs:
+            store = cast(services.StoreService, self._services.store)
+            libraries_md = store.get_libraries_metadata(installable_libs)
```

Comment:
note: ensure auth is not required

Reply:
This was already done in the spread test, but I've ensured that `CHARMCRAFT_AUTH` is unset in the spread test now with a comment as to why so we don't accidentally delete it.